logstash-core/lib/logstash/runner.rb

```diff
@@ -310,9 +310,17 @@ def execute
     if setting("config.debug") && !logger.debug?
       logger.warn("--config.debug was specified, but log.level was not set to \'debug\'! No config info will be logged.")
     end
-    if setting("pipeline.buffer.type") != nil
-      configure_pipeline_buffer_type
+    if setting("pipeline.buffer.type") == nil
+      deprecation_logger.deprecated(
+        "'pipeline.buffer.type' setting is not explicitly defined."\
+        "Before moving to 9.x set it to 'heap' and tune heap size upward or set it to 'direct' to maintain existing behavior."
```

Comment:
If we separate the two options by a comma, we more clearly delineate the (`set it to 'heap' and tune heap size upward`) option from the (`set it to 'direct' to maintain existing behavior`)

```suggestion
        "Before moving to 9.x set it to 'heap' and tune heap size upward, or set it to 'direct' to maintain existing behavior."
```